ENH Adds feature_name_combiner to OneHotEncoder (#22506)
scikit-learn/scikit-learn: doc/whats_new/v1.1.rst

:mod:`sklearn.preprocessing`
............................

- |Enhancement| Adds a `feature_name_combiner` parameter to :class:`preprocessing.OneHotEncoder`.

>   This specifies a custom callable to create feature names out with arguments:
>   (input_feature, category).

Reviewer: ```suggestion
  This specifies a custom callable to create feature names to be returned by :meth:`get_feature_names_out`.
  The callable combines input arguments `(input_feature, category)` to a string.
```

>   This allows specifiying a custom callable to create feature names out of (encoded_feature_name, category)
>   tuple.

Reviewer 2: I would can avoid the word "tuple" here:

```suggestion
  This specifies a custom callable to create feature names out with arguments:
  (input_feature, category).
```
